src/impl_pipeline.cpp

```diff
@@ -553,8 +566,14 @@ namespace daxa
 
         DAXA_DBG_ASSERT_TRUE_M(pipeline_result == VK_SUCCESS, "failed to create graphics pipeline");
 
-        vkDestroyShaderModule(impl.impl_device.as<ImplDevice>()->vk_device, v_vk_shader_module, nullptr);
-        vkDestroyShaderModule(impl.impl_device.as<ImplDevice>()->vk_device, p_vk_shader_module, nullptr);
+        auto destroy_shader_module = [&](VkShaderModule& vk_shader_module) {
```

Comment:
When you add a vector for shader modules, like suggested above, you can just loop over it to destroy them, no need for lambda.